src/beam.ts

```diff
@@ -10,8 +10,44 @@ import { Element } from './element';
 import { Fraction } from './fraction';
 import { Tuplet } from './tuplet';
 import { Stem } from './stem';
+import { Note } from './note';
+import { StemmableNote } from './stemmablenote';
+import { Voice } from './voice';
+import { RenderContext } from './types/common';
+
+export interface BeamInfo {
+  start: number;
+  end?: number;
+}
+
+export interface BeamRenderOptions {
+  flat_beam_offset?: number;
+  flat_beams: boolean;
+  secondary_break_ticks?: number;
+  show_stemlets: boolean;
+  beam_width: number;
+  max_slope: number;
+  min_slope: number;
+  slope_iterations: number;
+  slope_cost: number;
+  stemlet_extension: number;
+  partial_beam_length: number;
+  min_flat_beam_offset: number;
+}
+
+export interface GenerateBeamConfig {
```

Comment:
This can be inlined.

Reply:
Do you mean to inline GenerateBeamConfig? The structure has 9 attributes.

Reply:
Yes, correct -- the structure as a whole is only used once, so I prefer to keep it near where it's used.